File: php/src/Model/SupervisorIdentity.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * The supervisor a session belongs to.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class SupervisorIdentity
{
    public function __construct(
        public readonly string $id,
        public readonly string $label,
        public readonly ?string $email,
        /** Null for a workspace-wide supervisor. */
        public readonly ?string $teamId,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            label: (string) ($data['label'] ?? ''),
            email: isset($data['email']) ? (string) $data['email'] : null,
            teamId: isset($data['teamId']) ? (string) $data['teamId'] : null,
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/SupervisorIdentity.java
package io.fivexer.sdk.model;

/**
 * The supervisor a session belongs to.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class SupervisorIdentity {
    private String id;
    private String label;
    private String email;
    private String teamId;

    public String getId() { return id; }
    public String getLabel() { return label; }
    public String getEmail() { return email; }
    public String getTeamId() { return teamId; }
}
